src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,454 @@
+   */
+  tooltipProps?: Partial<Omit<TooltipProps, 'content' | 'children'>>;
+  /**
+   * Optional tooltip function that can be supplied to generate tooltip text.
+   */
+  tooltipRenderer?: (val: number, step: number) => React.ReactNode;
+  /**
+   *  Optional array of labels that will be used to determine number of ticks
+   * displayed and their labels. Ticks are spaced evenly across width of Slider.
+   */
+  tickLabels?: React.ReactNode[];
+  /**
+   * Optional label for the minimum value. If undefined then the min
+   * value is shown. Use empty string for no label.
+   */
+  minLabel?: React.ReactNode;
+  /**
+   * Optional label for the maximum value. If undefined then the max
+   *  value is shown. Use empty string for no label.
+   */
+  maxLabel?: React.ReactNode;
+  /**
+   * Defines the allowed behavior when moving Thumbs when multiple Thumbs are
+   * shown. It controls if a Thumb movement should be limited to only move in
+   * the segments adjacent to the Thumb. Possible values:
+   * 'allow-crossing' - allows thumb to cross other thumbs. Default.
+   * 'inhibit-crossing'- keeps the thumb from crossing and separated by a step.
+   * @default 'inhibit-crossing'
+   */
+  thumbMode?: 'allow-crossing' | 'inhibit-crossing';
+  /**
+   * Additional props to include on each thumb container.
+   */
+  thumbProps?: Omit<CommonProps, 'title'>;
+  /**
+   * Callback fired when the value(s) of the slider has changed. This will receive
+   * changes at the end of a slide as well as changes from clicks on rails and tracks.
+   */
+  onChange?: (values: ReadonlyArray<number>) => void;
+  /**
+   * Callback fired when the value(s) of the slider are internally updated during
+   * operations like dragging a Thumb. Use this callback with caution as a
+   * high-volume of updates will occur when dragging.
+   */
+  onUpdate?: (values: ReadonlyArray<number>) => void;
+} & Omit<CommonProps, 'title'>;
+
+/**
+ * Slider component that display Thumbs for each value specified along a Rail.
+ * @example
+ * <Slider values={[10]} min={0} max={60} disabled />
+ * <Slider values={[10, 20]} min={0} max={50} step={2} />
+ * <Slider values={[10, 20, 30, 40]} min={0} max={60} setFocus
+ *   thumbMode='allow-crossing' />
+ */
+export const Slider = React.forwardRef<HTMLDivElement, SliderProps>(
+  (props, ref) => {
+    const {
+      min = 0,
+      max = 100,
+      values,
+      step = 1,
+      setFocus = false,
+      tooltipProps,
+      tooltipRenderer,
+      disabled = false,
+      tickLabels,
+      minLabel,
+      maxLabel,
+      trackDisplayMode = undefined,
+      thumbMode = 'inhibit-crossing',
+      onChange,
+      onUpdate,
+      thumbProps,
+      className,
+      ...rest
+    } = props;
+
+    const [currentValues, setCurrentValues] = React.useState(values);
+    React.useEffect(() => {
+      setCurrentValues(values);
+    }, [values]);
+
+    const [minValueLabel, setMinValueLabel] = React.useState(minLabel);
```

Comment:
Thinking if the initial value shouldnt be this `minLabel ?? min.toString()`. `useEffect` will run eventually, but.. cannot determine the time for it. Not sure..

Reply:
Aren't hooks executed in order?

Reply:
I thought they were executed in order, which is one of that reasons they can't be conditional.

Reply:
I mean..
`useState` hook will run first, then it will be a render, then `useEffect` will be run and then it will rerender.
Hooks are executed in order, but `useEffect `is run after render, not before (like `useState`).

Reply:
It should not re-render if the `values` array in setState initializer matches `values` in setCurrentValue method.

Reply:
so `minLabel` porp might be not passed, it will be undefined as initial value.. in `useEffect` it will be assigned to `min.toString()` which gonna cause rerender.. Or I do not understand here something ?

Reply:
I guess I need to quit looking at the top of the block of code being shown.

Reply:
Updated to initialize the values using same logic so it avoids possible initial re-render.